src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -67,6 +64,211 @@ case class Task(
 }
 
 object Task {
+  private[this] def newHealthStatus(
+    current: MesosProtos.TaskStatus,
+    update: MesosProtos.TaskStatus): Option[MesosProtos.TaskStatus] = {
+
+    val healthy = update.hasHealthy && (!current.hasHealthy || current.getHealthy != update.getHealthy)
+    val changed = healthy || current.getState != update.getState
+    if (changed) {
+      Some(update)
+    }
+    else {
+      None
+    }
+  }
+
+  /**
+    * A LaunchedEphemeral task is a stateless task that does not consume reserved resources or persistent volumes.
+    */
+  case class LaunchedEphemeral(
+      taskId: Task.Id,
+      agentInfo: AgentInfo,
+      appVersion: Timestamp,
+      status: Status,
+      networking: Networking) extends Task {
+
+    private[this] val log = LoggerFactory.getLogger(getClass)
+
+    override val reservationWithVolumes: Option[ReservationWithVolumes] = None
+
+    override val launched: Option[Launched] = Some(Task.Launched(appVersion, status, networking))
+
+    private[this] def hasStartedRunning: Boolean = status.startedAt.isDefined
+
+    override def update(update: TaskStateOp): TaskStateChange = update match {
+      // case 1: now running
+      case TaskStateOp.MesosUpdate(MarathonTaskStatus.Running(mesosStatus), now) if !hasStartedRunning =>
+        val updated = copy(
+          status = status.copy(
+            startedAt = Some(now),
+            mesosStatus = mesosStatus))
+        TaskStateChange.Update(updated)
+
+      // case 2: terminal
+      case TaskStateOp.MesosUpdate(MarathonTaskStatus.Terminal(_), now) =>
+        TaskStateChange.Expunge
+
+      // case 3: health updated
+      case TaskStateOp.MesosUpdate(taskStatus, now) =>
```

Comment:
What if the `MesosState` changed? I think that the assumption here is that the task is running and stays running. That might or might not be always the case. Can we make it more obvious? At least the assumption?